addons/web/models/models.py

```diff
@@ -271,9 +276,20 @@ def group_id_name(value):
                     if parent:
                         parent_id = parent[0]
                         values['parent_id'] = parent_id
-                        counters[parent_id] = counters.get(parent_id, 0) + counters.get(record_id, 0)
+                        forest[parent_id]['children_ids'].append(record_id)
+                        forest[record_id]['is_root'] = False
                 category_values.append(values)
 
+            root_ids = [rec[0] for rec in forest.items() if rec[1]['is_root']]
+            forest['adjoint_id']['children_ids'] = root_ids
```

Comment:
adjoint ? isn't it the root of roots ?

Reply:
It is like a virtual node that transform the forest (many disjoint trees) into an actual tree, this in order to have a nice function compute_global_count